### ISSUE TITLE
Android: Example on Native Moduls wrong

### ISSUE THREAD

**Issue by Stophface** [facebook/react-native-website#470]
Following [this](https://facebook.github.io/react-native/docs/native-modules-android.html) example I encountered various errors in your code which makes the example not usable. 

[Number one](https://facebook.github.io/react-native/docs/native-modules-android.html#the-toast-module)

```
// ToastModule.java

package com.facebook.react.modules.toast;

import android.widget.Toast;

import com.facebook.react.bridge.NativeModule;
[...]
```

That won't work because the package (`package com.facebook.react.modules.toast;`) of each developer following the example should be `package com.your-app-name;` and not the `com.facebook` package.

[Number two](https://facebook.github.io/react-native/docs/native-modules-android.html#register-the-module)

```
// CustomToastPackage.java

package com.facebook.react.modules.toast;

import com.facebook.react.ReactPackage;
import com.facebook.react.bridge.NativeModule;
[...]
```

Consequently the package name is wrong here too (`package com.facebook.react.modules.toast;`). That should be `package com.your-app-name;`. Further, you are not importing the `ToastModule` Class. You are using it `modules.add(new ToastModule(reactContext));`, but you never imported it. This line needs to be in the import statements too `import com.your-app-name.ToastModule;`

[Finally](https://facebook.github.io/react-native/docs/native-modules-android.html#register-the-module), in the `MainApplication.java`, you write only this

```
protected List<ReactPackage> getPackages() {
    return Arrays.<ReactPackage>asList(
            new MainReactPackage(),
            new CustomToastPackage()); // <-- Add this line with your package name.
}
```

You fail to mention that the `CustomToastPackage` class package must be imported too. So this line `import com.your-app-name.CustomToastPackage;` should go into the import statements in `MainApplication.java`

**Comment by charpeni** [comment]
Thank you for bringing this!

Are you up to submit a PR? Your help would be greatly appreciated.

**Comment by Stophface** [comment]
@charpeni sure

**Comment by Stophface** [comment]
@charpeni I made two pull requests. One for `native-modules` and one for `native-ui-components`.  The changes in the `native-ui-components` are rather big as the documentation was just simply wrong or very(!) hard to follow. I hope the changes are to your standards.

**Comment by charpeni** [comment]
Your two PRs have been merged, let's close this issue. 

Let me know if you think it should remain open.
